File: apple/Sources/CCDAEngine/Models/CCDACodedValue.swift
import Foundation

/// Coded clinical value with optional display and code-system metadata.
public struct CCDACodedValue: Hashable, Sendable {
    /// Code value from the source XML.
    public let code: String?
    /// Code-system OID or identifier.
    public let codeSystem: String?
    /// Human-readable code-system name when present.
    public let codeSystemName: String?
    /// Human-readable display value for the code.
    public let displayName: String?

    /// Creates a coded clinical value.
    public init(
        code: String? = nil,
        codeSystem: String? = nil,
        codeSystemName: String? = nil,
        displayName: String? = nil
    ) {
        self.code = code
        self.codeSystem = codeSystem
        self.codeSystemName = codeSystemName
        self.displayName = displayName
    }
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Common/CCDACodedValue.kt
package com.ccdakit.engine

/**
 * Coded clinical value from a CDA element such as `code`, `value`, or `confidentialityCode`.
 *
 * @property code Code value from the source XML.
 * @property codeSystem OID or identifier for the coding system.
 * @property codeSystemName Human-readable coding system name.
 * @property displayName Human-readable display text supplied by the source XML.
 */
data class CCDACodedValue(
    val code: String? = null,
    val codeSystem: String? = null,
    val codeSystemName: String? = null,
    val displayName: String? = null,
)
